sw/device/lib/dif/dif_pinmux.c

```diff
@@ -5,19 +5,187 @@
 #include "sw/device/lib/dif/dif_pinmux.h"
 
 #include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/mmio.h"
 #include "sw/device/lib/dif/dif_base.h"
 
 #include "pinmux_regs.h"  // Generated.
 
-OT_WARN_UNUSED_RESULT
+static bool dif_pinmux_get_reg_offset(dif_pinmux_index_t index,
+                                      dif_pinmux_pad_kind_t kind,
+                                      ptrdiff_t mio_reg_base,
+                                      ptrdiff_t dio_reg_base,
+                                      ptrdiff_t *reg_offset) {
+  uint32_t num_pads;
+  ptrdiff_t reg_base;
+
+  switch (kind) {
+    case kDifPinmuxPadKindMio:
+      num_pads = PINMUX_PARAM_N_MIO_PADS;
+      reg_base = mio_reg_base;
+      break;
+    case kDifPinmuxPadKindDio:
+      num_pads = PINMUX_PARAM_N_DIO_PADS;
+      reg_base = dio_reg_base;
+      break;
+    default:
+      return false;
+  }
+  if (index >= num_pads) {
+    return false;
+  }
+  *reg_offset = reg_base + index * sizeof(uint32_t);
+  return true;
+}
+
+static bool dif_pinmux_get_sleep_status_bit(dif_pinmux_pad_kind_t kind,
+                                            dif_pinmux_index_t index,
+                                            ptrdiff_t *reg_offset,
+                                            bitfield_bit32_index_t *bit) {
+  uint32_t num_pads;
+  ptrdiff_t reg_base;
+
+  switch (kind) {
+    case kDifPinmuxPadKindMio:
+      num_pads = PINMUX_PARAM_N_MIO_PADS;
+      reg_base = PINMUX_MIO_PAD_SLEEP_STATUS_REG_OFFSET;
```

Comment:
For some reason, this generates differently between the meson and bazel build systems.

Reply:
Ah, different header dependency. One grabs from top_earlgrey; the other, from a dangling template output in the unassociated hw/ip/pinmux directory.